In [3]:
from utils import data_utils as du

Major city comparisons mainly use SF(2011), Chicago, New York(2009), DC data(2008).

In [49]:
cities = du.load_major_us_cities()
sf_df = cities["sf_Existing_Buildings_Energy_Performance_Ordinance_Report_20260114"]
dc_df = cities["dc_Building_Energy_Performance"]
nyc_df = cities["NYC_Municipal_Building_Energy_Benchmarking_Results_20260114"]
eva_df = cities[
    "Evanston_Energy_and_Water_Benchmarking_for_Evanston_IL_(2019_Data_Reported_in_2020)_1451443620066714074"
]
seattle_df = cities[
    "Seattle_Benchmarking_Performance_Ranges_by_Building_Type_2015-Present_20260125"
]

/project/src/utils/data_utils.py:679: DtypeWarning: Columns (4,23,24,44) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(file)
2026-01-26 01:24:10,857 [INFO] Loaded Seattle_Benchmarking_Performance_Ranges_by_Building_Type_2015-Present_20260125.csv → (34699, 46)
2026-01-26 01:24:10,861 [INFO] Loaded Evanston_Energy_and_Water_Benchmarking_for_Evanston_IL_(2019_Data_Reported_in_2020)_1451443620066714074.csv → (324, 37)
2026-01-26 01:24:10,872 [INFO] Loaded dc_Building_Energy_Performance.csv → (1816, 38)
2026-01-26 01:24:10,880 [INFO] Loaded NYC_Municipal_Building_Energy_Benchmarking_Results_20260114.csv → (3388, 20)
2026-01-26 01:24:10,979 [INFO] Loaded sf_Existing_Buildings_Energy_Performance_Ordinance_Report_20260114.csv → (28243, 34)


In [8]:
print(dc_df.columns)

Index(['X', 'Y', 'OBJECTID', 'PID', 'SSL', 'PMPROPERTYID', 'UBID',
       'PROPERTYNAME', 'PMPARENTPROPERTYID', 'PARENTPROPERTYNAME',
       'ADDRESSOFRECORD', 'OWNEROFRECORD', 'WARD', 'REPORTEDADDRESS', 'CITY',
       'STATE', 'POSTALCODE', 'YEARBUILT', 'PRIMARYPROPERTYTYPE_EPACALC',
       'TAXRECORDFLOORAREA', 'REPORTEDBUILDINGGROSSFLOORAREA', 'BEPS',
       'BEPS_METRIC_TYPE', 'BEPS_PROPERTY_GROUP', 'PROPERTY_BEPS_METRIC_YEAR',
       'PROPERTY_BEPS_METRIC', 'MEETS_BEPS', 'DISTANCE_FROM_BEPS_ESTIMATED',
       'PERFORMANCE_REQUIREMENT_EST', 'LATITUDE', 'LONGITUDE', 'XCOORD',
       'YCOORD', 'MAR_ID', 'GIS_ID', 'GLOBALID', 'LASTUPDATE',
       'APPROVED_COMPLIANCE_PATHWAY'],
      dtype='object')


### DC
DC has 1508 rows
Contains BEPS (Building Energy Performance Standards based on Site EUI, Energy Star Score, and YoY changes).

In [25]:
dc_df_cleaned = dc_df[dc_df["BEPS_METRIC_TYPE"] == "ENERGYSTARSCORE"]
dc_df_cleaned["BEPS"]
dc_df_cleaned["PID"].nunique()

1508

### SF
SF has 28,243 rows.
Contains ENERGY STAR Score, Site EUI, Source EUI, GHG metrics, and fuel usage (2011 - 2024).

In [43]:
print(sf_df["Building Name"].nunique())

sf_df_concu_num = sf_df["Building Name"].value_counts().eq(14).sum()
sf_df_concu_num

2496


1523

### NYC
NYC has 3388 rows. 
Does not contain ENERGY STAR Score, but does include Source EUI and GHG intensity for multiple years (2010–2013).

In [36]:
nyc_df.head(5)

,Borough,Block,Lot,BIN,Building,Agency,2010 Rating *,2010 Source EUI (kBtu/sq.ft.),2010 GHG Emissions Intensity (kgCO2e/ft²) †,2011 Rating *,2011 Source EUI (kBtu/sq.ft.),2011 GHG Emissions Intensity (kgCO2e/ft²) †,2012 Rating *,2012 Source EUI (kBtu/sq.ft.),2012 GHG Emissions Intensity (kgCO2e/ft²) †,2013 Rating *,2013 Source EUI (kBtu/sq.ft.),2013 GHG Emissions Intensity (kgCO2e/ft²) †,EPAPMBenchmarkedAs,Campus Name
0,2,2383,12,2001505,1332 Fulton Ave D C C Inc,ACS,N/A,180.6,4.9,NaN,155.9,4.3,NaN,157.7,4.3,NaN,164.8,4.5,Individual building,NaN
1,1,93,1,1062652,ACS Administrative Site,ACS,57,239.0,6.2,55,242.1,6.3,59,228.3,5.9,64.0,217.9,5.6,Individual building,NaN
2,3,4484,13,3098733,ACS Administrative Site,ACS,16,286.6,8.6,19,268.3,7.9,20,258.6,7.6,24.0,252.1,7.5,Space within building,NaN
3,2,4101,1,2044743,ACS Bronx Field Office,ACS,7,316.2,8.9,14,269.6,7.2,12,273.0,7.4,12.0,282.8,7.9,Individual building,NaN
4,3,2022,18,3057913,Administration Site,ACS,21,313.6,9.8,31,272.4,8.4,35,257.9,7.5,37.0,256.1,7.6,Individual building,NaN


In [41]:
print(nyc_df["Building"].nunique())

nyc_df_concu_num = nyc_df["Building"].value_counts().eq(1).sum()
nyc_df_concu_num

3273


3248

### Evanston
Evanston has 324 rows. 
Includes Energy Star Score, Site/Source EUI etc. (2018-2019)

### Seattle
Evanston has 34,699 rows. 
Includes Energy Star Score, Site/Source EUI etc. (2015-Present)

In [51]:
seattle_df.head()

,OSEBuildingID,DataYear,BuildingName,BuildingType,TaxParcelIdentificationNumber,Address,City,State,ZipCode,Latitude,...,Electricity(kWh),SteamUse(kBtu),NaturalGas(therms),ComplianceStatus,ComplianceIssue,Electricity(kBtu),NaturalGas(kBtu),TotalGHGEmissions,GHGEmissionsIntensity,Demolished
0,1,2024,MAYFLOWER PARK HOTEL,NonResidential,659000030,405 OLIVE WAY,SEATTLE,WA,98101,47.61220,...,"1,045,040","1,949,686","15,986",Not Compliant,Default Data,"3,565,676","1,598,590",263.3,2.98,False
1,2,2024,PARAMOUNT HOTEL,NonResidential,659000220,724 PINE ST,SEATTLE,WA,98101,47.61307,...,"787,838",NaN,"36,426",Compliant,No Issue,"2,688,104","3,642,560",208.6,2.36,False
2,3,2024,WESTIN HOTEL (Parent Building),NonResidential,659000475,1900 5TH AVE,SEATTLE,WA,98101,47.61367,...,"11,279,080","23,256,386","58,726",Compliant,No Issue,"38,484,221","5,872,650","2,418.2",3.18,False
3,5,2024,HOTEL MAX,NonResidential,659000640,620 STEWART ST,SEATTLE,WA,98101,47.61412,...,"796,976","1,389,935","11,648",Compliant,No Issue,"2,719,282","1,164,760",190.1,3.1,False
4,8,2024,WARWICK SEATTLE HOTEL,NonResidential,659000970,401 LENORA ST,SEATTLE,WA,98121,47.61395,...,"1,349,025",NaN,"73,811",Compliant,No Issue,"4,602,873","7,381,080",417.9,3.68,False


In [55]:
print(seattle_df["BuildingName"].nunique())

seattle_df_concu_num = seattle_df["BuildingName"].value_counts().eq(10).sum()
seattle_df_concu_num

3785


3013